src/main/java/be/seeseemelk/mockbukkit/potion/MockPotionEffectType.java

```diff
@@ -39,13 +58,18 @@ public class MockPotionEffectType extends PotionEffectType
 	 */
 	public MockPotionEffectType(@NotNull NamespacedKey key, int id, String name, boolean instant, Color color)
 	{
-		super(id, key);
+		this(key, id, name, instant, color, Category.NEUTRAL);
+	}
 
-		this.id = id;
-		this.name = name;
-		this.instant = instant;
-		this.color = color;
-		this.attributeModifiers = new EnumMap<>(Attribute.class);
+	public MockPotionEffectType(JsonObject data)
+	{
+		this(Objects.requireNonNull(NamespacedKey.fromString(data.get("key").getAsString())),
```

Comment:
Is the `Objects.requireNonNull` necessary?

Reply:
Nope, it is not, It's getting checked in the targeted constructor later on